node/src/chain/fsm.rs

```diff
@@ -863,18 +884,38 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network>
 
             // Try to accept other consecutive blocks from the pool, if
             // available
-            for height in (h + 1)..(self.range.1 + 1) {
+            for height in self.range.0..=self.range.1 {
                 if let Some(blk) = self.pool.get(&height) {
                     acc.try_accept_block(blk, false).await?;
+                    self.range.0 += 1;
+                    debug!(
+                        event = "accepting next block",
+                        block_height = height,
+                        last_request = self.last_request,
+                        mode = "out_of_sync"
+                    );
                 } else {
+                    self.request_missing_block(height).await;
```

Comment:
I'm not really convinced by this.

A missing block here might be in transit. 

I think the timeout-based `request_missing_blocks` should be sufficient.

Reply:
With the new getResource messages are sent sequentially, so for sure the missing block is not in transit

Reply:
Since it's UDP, being sent sequentially, doesn't mean they arrive in the correct order, does it?